src/utils.rs

```diff
@@ -52,6 +52,23 @@ pub fn do_write_line(
     }
 }
 
+pub fn get_modification_time(file_path: &str) -> String {
```

Comment:
I don't get this, can you explain a bit more? 😅

Reply:
Mocking means replacing the implementation of external functions/objects your code is calling in order to ensure a controlled test environment.

So for example, mocking `fs::metadata()` would allow you to provide an alternative implementation (only in the context of the unit tests) that might return a hardcoded modification time for a given file path, or return an error for another given file path (without relying on those file paths representing actual files on disk).
After a bit of research though, it seems mocking `fs::metadata()` isn't really doable. However I came across [std::fs::File::set_modified](https://doc.rust-lang.org/stable/std/fs/struct.File.html#method.set_modified) which would allow you to set the modification time of a temp file to an arbitrary value, which you can test against later in your test.
For testing the error case, you'll need to create a temp file that isn't readable.

I'll leave mocking the fallback to `SystemTime::now()` as an exercise :)

Reply:
Got it! Thanks

Reply:
I updated the MR!